### ISSUE TITLE
bug: `to_numpy()` differs between pandas and Polars for timezone-aware Series

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#1299]
```python
import pandas as pd
import polars as pl
from datetime import datetime

df = pl.DataFrame({'price': [40, 50], 'datetime': [datetime(2020,1,1), datetime(2020,1,2)]})
df = df.with_columns(pl.col('datetime').dt.replace_time_zone('Asia/Kathmandu'))
print(df['datetime'].to_numpy())
print(df['datetime'].to_pandas().to_numpy())
```
in one case, it converts to UTC, in the other we can object array of Timestamp objects

I'd prefer to avoid `object` dtype as much as possible, so I'd advocate for the former. If somebody wants the local datetime, they can use `.dt.replace_time_zone(None)`

```
In [7]: nw.from_native(df['datetime'], series_only=True).to_numpy()
Out[7]: 
array(['2019-12-31T18:15:00.000000', '2020-01-01T18:15:00.000000'],
      dtype='datetime64[us]')

In [8]: nw.from_native(df['datetime'].to_pandas(), series_only=True).to_numpy()
Out[8]: 
array([Timestamp('2020-01-01 00:00:00+0545', tz='Asia/Kathmandu'),
       Timestamp('2020-01-02 00:00:00+0545', tz='Asia/Kathmandu')],
      dtype=object)
```

**Comment by FBruzzesi** [comment]
Closed by #1305 ?

**Comment by MarcoGorelli** [comment]
yup, thanks!
